File: backend/src/main/kotlin/com/example/backend/dtos/InventaireDto.kt
package com.example.backend.dtos


data class ProduitInventaireDto(
  val produitId: Long,
  val rayonId: Long,
  val quantiteReel: Int,
  val quantiteSysteme: Int,
)

data class InventaireRequestDto(
  val dateDeDebut: String,
  val produitList: List<ProduitInventaireDto>,
)

data class InventaireNewCreatetDto(
  val rayonId: String?,
  val categorieId: String?,
  val fabriquantId: String?,
  val formeId: String?,
  val fournisseurId: String?,
)

data class InventaireUpdateRequestDto(
  val id: String,
  val produitList: List<ProduitInventaireDto>,
)

data class InventaireOneProductUpdateRequestDto(
  val id: String?,
  val produitId: Long?,
  val rayonId: Long?,
  val quantiteReel: Int?,
  val quantiteSysteme: Int?,
  val isValid: Boolean? = false,
)

File: backend_py/pharmaPython/app/schemas/inventaire_dto.py
from pydantic import BaseModel
from typing import Optional, List

class ProduitInventaireDto(BaseModel):
  produitId: int
  rayonId: int
  quantiteReel: int
  quantiteSysteme: int

class InventaireRequestDto(BaseModel):
  dateDeDebut: str
  produitList: List[ProduitInventaireDto]

class InventaireNewCreatetDto(BaseModel):
  rayonId: Optional[str] = None
  categorieId: Optional[str] = None
  fabriquantId: Optional[str] = None
  formeId: Optional[str] = None
  fournisseurId: Optional[str] = None

class InventaireUpdateRequestDto(BaseModel):
  id: str
  produitList: List[ProduitInventaireDto]

class InventaireOneProductUpdateRequestDto(BaseModel):
  id: Optional[str] = None
  produitId: Optional[int] = None
  rayonId: Optional[int] = None
  quantiteReel: Optional[int] = None
  quantiteSysteme: Optional[int] = None
  isValid: Optional[bool] = False
